### ISSUE TITLE
Torch backend: convert_to_tensor creates float64 constants in torch.export graphs when dtype='float32' is requested

### ISSUE THREAD

**Issue by pctablet505** [keras-team/keras#22815]
## Bug Description

In the PyTorch backend, `keras.ops.convert_to_tensor(scalar_float, dtype="float32")` incorrectly produces `float64` constants in `torch.export` graphs. This happens because the function internally converts Python scalar floats to `np.array(float)` (which defaults to `float64`) before passing to `torch.as_tensor`. During `torch.export`, the original `float64` numpy array gets lifted as a constant *before* the cast to `float32` occurs in the graph.

This causes downstream MLIR verification errors in strict conversion pipelines (e.g. `litert_torch.convert`), which reject dtype mismatches between operation operands.

## Standalone code to reproduce

```python
import os
os.environ["KERAS_BACKEND"] = "torch"

import torch
import keras

class ModuleWithScalarConvert(torch.nn.Module):
    def forward(self, x):
        # This is what happens inside ops like maximum() when a Python
        # scalar (e.g. backend.epsilon()) is used alongside a float32 tensor.
        scalar = keras.ops.convert_to_tensor(1e-7, dtype="float32")
        return torch.maximum(x, scalar)

model = ModuleWithScalarConvert()
exported = torch.export.export(model, (torch.ones(5, dtype=torch.float32),))

print("Exported constants:")
for name, val in exported.constants.items():
    if isinstance(val, torch.Tensor):
        print(f"  {name}: dtype={val.dtype}, shape={val.shape}")
```

**Actual output:**
```
  lifted_tensor_0: dtype=torch.float64, shape=torch.Size([])
```

**Expected output:**
```
  lifted_tensor_0: dtype=torch.float32, shape=torch.Size([])
```

## Impact

Any layer that mixes Python scalar floats with float32 tensors via backend ops will silently inject `float64` constants into exported graphs. This was first observed with `layers.Normalization` + `litert_torch.convert`, but the root cause is in the core torch backend and could affect any `torch.export` workflow.

## Proposed Fix

In `keras/src/backend/torch/core.py`, skip `np.array()` for scalar Python values (`bool`, `int`, `float`) so they go directly to `torch.as_tensor()` with the requested dtype:

```diff
-    if not isinstance(x, (list, tuple)):
+    if isinstance(x, (list, tuple)):
+        if len(x) > 0 and any(isinstance(x1, torch.Tensor) for x1 in x):
+            return torch.stack([convert_to_tensor(x1) for x1 in x])
+    elif not isinstance(x, (bool, int, float)):
         x = np.array(x)
-    elif len(x) > 0 and any(isinstance(x1, torch.Tensor) for x1 in x):
-        return torch.stack([convert_to_tensor(x1) for x1 in x])
```

## System information

- Keras version: master (commit ~4caa4a087)
- Backend: torch
- PyTorch version: 2.11.0
- Python version: 3.13.7
- OS: Linux
